internal/dag/gatewayapi_processor.go

```diff
@@ -847,12 +855,17 @@ func (p *GatewayAPIProcessor) routes(matchConditions []*matchConditions, headerP
 
 // cluster builds a *dag.Cluster for the supplied set of headerPolicy and service.
 func (p *GatewayAPIProcessor) cluster(headerPolicy *HeadersPolicy, service *Service, weight uint32) *Cluster {
-	return &Cluster{
+	cluster := &Cluster{
 		Upstream:             service,
 		Weight:               weight,
 		Protocol:             service.Protocol,
 		RequestHeadersPolicy: headerPolicy,
 	}
+	if len(p.MinimumTLSVersion) != 0 {
+		cluster.MinimumProtocolVersion = p.MinimumTLSVersion
+		cluster.CipherSuites = p.CipherSuites
+	}
```

Comment:
ditto re: assigning each one independently based on whether it's configured on the `GatewayAPIProcessor`.